### PR TITLE
Fix for Halloween update

### PR DESCRIPTION
**Pull request by developer9998**
This fixes hats from not working as well as a few other issues that were in the mod.

image

Preview video: https://www.youtube.com/watch?v=jl8esnwWxPw&ab_channel=dev%27smainchannel.

UPDATE: The button bug has been fixed in the latest commit

### CODE REVIEW

**Comment by Graicc** [comment]
> Not sure if it was in the prior update, but your original cosmetic will still be equipped if you have a modded cosmetic when the game is open

You changed the method responsible for removing the original cosmetic

File: GorillaCosmetics/UI/SelectionManager.cs [hunk 1]
```diff
@@ -405,22 +424,62 @@ public void ResetMaterial()
...
-			offlineCustomCosmeticsController.ResetMaterial();
-			onlineCustomCosmeticsController?.ResetMaterial();
 
-			UpdateHeadModel();
+            offlineCustomCosmeticsController.ResetMaterial();
+
+            if (PhotonNetwork.InRoom)
+            {
+                if (onlineCustomCosmeticsController == null)
+                    onlineCustomCosmeticsController ??= GorillaTagger.Instance?.myVRRig?.GetComponent<ICustomCosmeticsController>(); // TODO: ew
```
Comment (Graicc) [inline GorillaCosmetics/UI/SelectionManager.cs]: `??=` does the null check implicitly

`onlineCustomCosmeticsController ??= GorillaTagger.Instance?.myVRRig?.GetComponent<ICustomCosmeticsController>();` really should only exist in one place, like how it was in the previous implementation

**Comment by developer9998** [comment]
still having issues with onlineCustomCosmeticsController and this error happens when you join a lobby sometimes: 
`[Error  : Unity Log] Error while updating player cosmetics for (USERNAME): System.NullReferenceException: Object reference not set to an instance of an object
  at GorillaCosmetics.CosmeticsNetworker.OnPlayerPropertiesUpdate (Photon.Realtime.Player targetPlayer, ExitGames.Client.Photon.Hashtable changedProps) [0x00019] in <68979d7e9d364e8a99bf91481e1dcd61>:0`

**Comment by developer9998** [comment]
Forgot to mention turns out latest commit fixes a lot of custom colour errors by (for the most part) changing up one line lmao

**Comment by sonc9r9843724839** [comment]
fork fork where is it

**Pull request closed without merging**